system/Boot.php

```diff
@@ -196,7 +196,14 @@ protected static function definePathConstants(Paths $paths): void
 
         // The path to the writable directory.
         if (! defined('WRITEPATH')) {
-            define('WRITEPATH', realpath(rtrim($paths->writableDirectory, '\\/ ')) . DIRECTORY_SEPARATOR);
+            if (($writePath = realpath(rtrim($paths->writableDirectory, '\\/ '))) === false) {
+                header('HTTP/1.1 503 Service Unavailable.', true, 503);
+                echo 'The WRITEPATH is not set correctly.';
+
+                // EXIT_ERROR is not yet defined
+                exit(1);
+            }
```

Comment:
```suggestion

            $writePath = realpath(rtrim($paths->writableDirectory, '\\/ '));
        
            if ($writePath === false) {
                header('HTTP/1.1 503 Service Unavailable.', true, 503);
                echo 'The WRITEPATH is not set correctly.';
                // EXIT_ERROR is not yet defined
                exit(1);
            }
```

But it looks good either way.

Reply:
I started with what you suggested and then decided to make it more condensed.

If more people find it more readable, I can make changes.

Reply:
I like datamweb's style.

Reply:
We talked about it - a separate variable is better
https://github.com/codeigniter4/CodeIgniter4/pull/9249#discussion_r1826596139

Reply:
Updated.

Reply:
@neznaika0 A quite different situation.